### PR TITLE
remove-tag-button-transition

### PR DESCRIPTION
**Pull request by johanstromqvist**
### WHY are these changes introduced?
Tag button hover state should be immediate.

### WHAT is this pull request doing?
Removes excessive hover state transition on tag button.

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {Page} from '@shopify/polaris';

interface State {}

export default class Playground extends React.Component<never, State> {
  render() {
    return (
      <Page title="Playground">
        {/* Add the code you want to test here */}
      </Page>
    );
  }
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)

### CODE REVIEW

**Comment by danrosenthal** [comment]
Hey @johanstromqvist, would you mind providing a bit of rationale for this change, and pinging a couple reviewers?

cc: @ry5n

**Comment by ry5n** [comment]
I’d want to understand the rationale as well.

**Comment by danrosenthal** [comment]
@johanstromqvist one more bump for context on this change

**Comment by johanstromqvist** [comment]
Sorry about the delay guys – just got back from vacation today. Expect rationale within the next couple of days, or let me know if it's urgent and blocking you. I'll also provide more rationale in future PRs.

**Comment by johanstromqvist** [comment]
Sorry about the delay.

The short story is that the touch target is behaving inconsistently and instead of fixing it I simply removed the behaviour. I asked around in the Polaris team but we couldn't find any rationale for the transition, and I couldn't see any myself. We could see tho that the component hadn't been updated in a long time.

**Why do I think transition is excessive**
The transition doesn't solve any problems or improve UX in my opinion, and I can see how adding transitions to hover states can add a sluggish delay which distances users from direct control.

**Why do I suggest hover to be immediate**
As a rule of thumb, I'd suggest that hover feedback should default to immediate, unless the hover triggers state change that would benefit from clarification. Before suggesting that as a general guideline, I'd like to try out on a couple of scenarios and run it by you guys. It's my current opinion tho.

**The inconsistency**
The current tag button has a 100-200ms transition on the background of the touch target and 0ms transition on the `x` icon in the centre, causing it to feel wonky and uncrafted.

File: UNRELEASED.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 - Updated open styleguide pr to create multiple pull requests to update `polaris-react` across multiple repos ([#1069](https://github.com/Shopify/polaris-react/pull/1069))
 - Updated the pull request creation to retry when it fails ([#1069](https://github.com/Shopify/polaris-react/pull/1069))
 - Exported overlay and layer data attributes for use in consumer components ([#1266](https://github.com/Shopify/polaris-react/pull/1266))
+- Removed transition on tag button hover state [#1337](https://github.com/Shopify/polaris-react/pull/1337)
```
Comment (tmlayton) [inline UNRELEASED.md]:
```suggestion
- Removed transition on tag button hover state ([#1337](https://github.com/Shopify/polaris-react/pull/1337))
```

**Pull request merged**